goalgorilla/open_social
# Issue #3120797: Invite to Event feature

## Problem
The event enrollment system currently is limited to a few options. To offer more flexibility we would like to add the follow options:
Request to enroll - When you to approve people before enrolling, this option allows people within or outside the group to request to enroll.
Invite - When you only want preselected members to enroll you can invite people to your event.

## Solution
Add more enrollment options and a submodule to invite people to events.

See also #1774.

## Issue tracker
https://www.drupal.org/project/social/issues/3120797

## How to test
Request to enroll
- [ ] By default you are now able to select 2 options for enrolling, request to enroll without enabling a submodule
- [ ] When choosing to request to enroll, if this event is within a group you are also able to allow people outside of the group. Try both of them.
- [ ] When going to the event it should show "request to enroll" instead of enroll. When clicking on it a lightbox will show up where you can fill in an additional message. After sending in the message, run the cron and login as an event manager
- [ ] As an event manager you should have a notification and email that someone requested an enrollment. When clicking on the notification it directs you to the event page.
- [ ] On the event page it show under the hero that 1 request is pending, click on it and see the overview with requests
- [ ] Accept or deny the request, when you now check the enrollment list you can see the user in the list.
- [ ] Login as the enrolled users and see you are now enrolled. Note that leaving the event will require you to request again.

Invite to enroll
- [ ] Enable the submodule social_event_invite
- [ ] When now going to an event, you see a third option is shown "invite only". 
- [ ] Select this and go to the manage enrollments page, you are now able to add members with 3 different routes.
- Directly
This will directly add members to the event like you would normally do.
- By email
This will send out an invitation mail where users can visit the event and decide to enroll
- Or view invites
Here you can see who you invited to the event
- [ ] Try out all three of them

## Release notes
We have added two new options to manage enrollments for your event.
- "Request to enroll" requires users to request an enrollment for an event that must be accepted by the manager before you are enrolled.
- "Invite" requires the event manager to manually invite people to an event by adding them directly or sending them an invite email.

Also an extra option has been added on top to allow people to use the above option if they are not a member yet of the group. This allows people to enroll to an event in a closed group without having to be part of the group.

Note that the request to enroll is enabled by default whereas invite is a submodule which needs to be enabled by enabling social_event_invite.

## Change Record
If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.

## Translations
